### DISCUSSION TITLE
How do you scrub through many Frigate videos?

### DISCUSSION THREAD

**Discussion by szaske** [blakeblackshear/frigate#1405] [Ask A Question]
Well last night someone broke into my car and Frigate missed it. It's very frustrating especially because my driveway is very well lit. It seems Frigate has a terrible time recognizing people at night. Anyway I'm going to enable full time recording at this point and wanted to make a few suggestions to improve Frigate.

1. Ability to full time record at night only. Frigate seems to work great during the day, but when I look at my event history there is usually a big gap starting at around 10pm until 6am the next day.
2. Better way to view all existing videos. It would be nice if Frigate could enable some sort of video player that shows a timeline and walks through both the full time recorded videos as well as the events. I think a lot of people enjoy the type of UI that Hikvision and Blue Iris provides. As it is searching through Frigates list is pretty frustrating. I have a lot of people who walk by my house, so over the course of a day I'll have a hundred or so videos listed. If you click on any of them it replaces the Frigate list with that video and when you return to the list Frigate resets the list back to the last x videos. This gets really tedious watching a hundred videos trying to find the guy breaking into my car. The best option is to click on the picture and say "Open in new Tab". Anyway This is by far the weakest aspect of Frigate, being able to scrub through previous days activity.

Let me know if I'm missing some Frigate UI that improves this or if anyone has a 3rd party way to quickly scrub through Frigate videos.  Even if I did have recording enabled I'm not looking forward to scrubbings through thousands of recording videos trying to find a thief (since Frigate recordings are only 1 minute long each)

**Comment by lawm** [comment]
I also have detection problems at night.  I've tried tweaking the settings, but it's still not very good.

As for viewing 24/7 recordings, there will be a new continuous video player in the next version:  https://github.com/blakeblackshear/frigate/pull/1155#issuecomment-854948555
It would be better if we could just rely on motion/object detection, though.

**Reply by szaske** [reply]
![frigate](https://user-images.githubusercontent.com/19757740/126267379-ff1640dd-4e09-4582-bd0c-66bbe65250b6.jpg)
Here's my driveway. It seems pretty bright to me. I do not use infrared as I believe the light is enough. Do you think  detection would be better WITH it?

**Reply by lawm** [reply]
Ok, that's different from my problem.  My problem is with black and white IR mode and low light color mode.

What I have done before to tune missed detection is:
* Save 24/7 recording files that include the missed event. 
* Combine into 1 longer file.
	for f in *.mp4; do echo "file '$f'"; done > list.txt
	ffmpeg -f concat -i  list.txt -c copy out.mp4
* Re-encode to match my substream resolution 640x480 6fps
	ffmpeg -i out.mp4 -vf scale=640:480 -r 6 out-substream.mp4
* Setup new instance in a vm
* Setup local file as source instead of actual camera
	like in https://github.com/blakeblackshear/frigate/blob/release-0.9.0/docs/docs/contributing.md#setup with -re framerate option 

Or, since you won't have recorded files of the missed event, you could just walk in front of your camera at night.

* Then check the camera view in Frigate and enable the motion boxes and bounding boxes.

See if it can detect motion or not.  I think it's the red boxes.  If not, adjust motion settings  ([advanced#motion](https://blakeblackshear.github.io/frigate/configuration/advanced#motion))
If it can detect motion, but not the person, adjust object detection settings.

In your case, it's probably detecting motion but not the person.  So, you'll want to lower min_score and threshold a little until people are detected:
https://blakeblackshear.github.io/frigate/configuration/false_positives 
defaults are at https://blakeblackshear.github.io/frigate/configuration/index#objects

If it's too low, you'll get false positives.

Some false positives can be countered with bigger min_area and other settings.


Edit: In short, you could just lower threshold from default 0.7 and see what happens.  You'll see how low 'Score' gets in the Events page.  For me, I see high 50%- 60% score, so I set threshold to around .5.
